osclib/stagingapi.py

```diff
@@ -807,8 +807,10 @@ def rm_from_prj(self, project, package=None, request_id=None,
 
         for sub_prj, sub_pkg in self.get_sub_packages(package):
             sub_prj = self.map_ring_package_to_subject(project, sub_pkg)
-            if sub_prj != subprj:  # if different to the main package's prj
-                delete_package(self.apiurl, sub_prj, sub_pkg, force=True, msg=msg)
+            # Skip inner-project links for letter stagings as they should
+            # already be in place, but for adi packages remove them.
+            if not self.is_adi_project(project) and sub_prj == project: continue
```

Comment:
I would expect `map_ring_package_to_subject()` to have no effect on an adi project given:
```python
if self.is_adi_project(project):
    return project
```
As such it should not matter, but seems better stylistically to check project.